salt/client/__init__.py

```diff
@@ -1641,7 +1641,7 @@ def _prep_pub(self,
         #   module
 
         # Generate the standard keyword args to feed to format_payload
-        payload_kwargs = {'cmd': 'publish',
+        payload_kwargs = {'cmd': kwargs.get('cmd', 'publish'),
```

Comment:
I'm not following the logic behind this. Could you please clarify?

Reply:
from the RFC here https://github.com/saltstack/salt/blob/develop/rfcs/0002-netapi-async-batch.md

> a new method is added to ClearFuncs that is called in place of the publish for asynchronous batched command execution (eg. publish_batch). This method will receive batch-specific parameters and use publish internally

The change you are asking about was done in that direction, meaning that it allows me to specify the publish function I want to use from the master. This is unsafe at the moment since it allows calling any function on the master. This problem will be addressed before merging the PR.